src/main/java/pl/edu/uj/cluster/DefaultDistributor.java

```diff
@@ -38,37 +39,49 @@
     private NodeList nodeList;
     @Autowired
     private MessageGateway messageGateway;
+    private AtomicBoolean workerPoolOverflow = new AtomicBoolean(false);
```

Comment:
nazwa nie mówi zabardzo o przeznaczeniu tego pola.
Na stan obecny wydaje mi się , że jest to onWorkerPoolOverflowIsDuringExecution.